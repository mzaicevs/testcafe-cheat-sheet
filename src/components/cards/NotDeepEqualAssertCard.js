import React from "react";
import { assert_notDeepEqual } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";

const NotDeepEqualAssertCard = () => (
  <CodeCard
    cardTitle={assert_notDeepEqual.title}
    fullDocLink={assert_notDeepEqual.fullDocLink}
    codeSnippet={{
      code: assert_notDeepEqual.code,
      language: assert_notDeepEqual.language,
      options: assert_notDeepEqual.options,
      example: assert_notDeepEqual.example
    }}
    description={assert_notDeepEqual.description}
  />
);

export default NotDeepEqualAssertCard;
