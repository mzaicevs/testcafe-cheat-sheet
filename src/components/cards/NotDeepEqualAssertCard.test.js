import React from "react";
import { shallow } from "enzyme";
import NotDeepEqualAssertCard from "./NotDeepEqualAssertCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("NotDeepEqualAssertCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<NotDeepEqualAssertCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
