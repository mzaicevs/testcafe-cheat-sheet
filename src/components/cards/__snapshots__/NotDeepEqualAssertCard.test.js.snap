// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotDeepEqualAssertCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Not Deep Equal - Assert"
  codeSnippet={
    Object {
      "code": Array [
        "t.expect( actual ).notEql( unexpected, message, options );",
      ],
      "example": "import { Selector } from 'testcafe';

  fixture \`My fixture\`;
  
  test('My test', async t => {
      await t
        .expect(Selector('.className').count).notEql(2)
        .expect({ a: 'bar' }).notEql({ a: 'bar' }, 'this assertion will fail')
        .expect({ a: 'bar' }).notEql({ a: 'foo' }, 'this assertion will pass');
  });",
      "language": "javascript",
      "options": "{ 
    timeout: Number,
      allowUnawaitedPromise: Boolean 
  }",
    }
  }
  description="Asserts that actual is not equal to expected."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/assertions/assertion-api.html#not-deep-equal"
/>
`;
